client/components/theme/index.jsx

```diff
@@ -172,8 +176,12 @@ const Theme = React.createClass( {
 								context: 'singular noun, the currently active theme'
 							} ) }</span>
 						}
-						{ price && ! purchased &&
-							<span className="theme-badge__price">{ price }</span>
+						{
+							config.isEnabled( 'jetpack/pijp' )
+							? ! this.props.isJetpack && price && ! purchased &&
```

Comment:
This will hide the price only for themes in Jetpack sites because users won't be individually purchasing the themes, unlike what they can do for wpcom sites.
In Jetpack sites, they will have access to all themes at once when they purchase a Professional plan.